src/crawlee/storages/_key_value_store.py

```diff
@@ -119,13 +119,17 @@ async def open(
         configuration = service_locator.get_configuration() if configuration is None else configuration
         storage_client = service_locator.get_storage_client() if storage_client is None else storage_client
 
+        client_opener = storage_client.create_kvs_client(id=id, name=name, alias=alias, configuration=configuration)
+        additional_cache_key = storage_client.get_additional_cache_key(configuration=configuration)
+
         return await service_locator.storage_instance_manager.open_storage_instance(
             cls,
             id=id,
             name=name,
+            client_opener=client_opener,
```

Comment:
This is really misleading name since you reworked it and it's not an opener (function) anymore. This should be more `client_coro` or something.